test/integration/framework/etcd.go

```diff
@@ -74,15 +73,6 @@ func startEtcd() (func(), error) {
 		os.Setenv("ETCD_UNSUPPORTED_ARCH", "arm64")
 	}
 
-	etcdURL = env.GetEnvAsStringOrFallback("KUBE_INTEGRATION_ETCD_URL", "http://127.0.0.1:2379")
```

Comment:
this looks like a behavioral change?

Reply:
bazel will perform parallel tests, and in each "go_test" rule, each test case will create a etcd server, and this server will be closed at the end of each test.
The original code will cause a random failure bug, if a case passed, the etcd server with 2037 port will be closed, then other parallel cases will be failed due to "err (dial tcp 127.0.0.1:2379: connect: connection refused)".
So I modified the code logic to avoid random failures,  in each "go_test" rule, it will create a etcd server with a random port.

Reply:
does this break `go test ./test/integration/<sometest>` use?

Reply:
> does this break `go test ./test/integration/<sometest>` use?

I don't think so.

Reply:
I'm not sure if I understand who sets `KUBE_INTEGRATION_ETCD_URL` specifically, it looks like we used to support _reading_ that and now only set it?

Reply:
Please see the code in "staging/src/k8s.io/apiextensions-apiserver/test/integration/fixtures/server.go:IntegrationEtcdServers()"
```
// IntegrationEtcdServers returns etcd server URLs.
func IntegrationEtcdServers() []string {
	if etcdURL, ok := os.LookupEnv("KUBE_INTEGRATION_ETCD_URL"); ok {
		return []string{etcdURL}
	}
	return []string{"http://127.0.0.1:2379"}
}
```
If we don't set KUBE_INTEGRATION_ETCD_URL, it will get the wrong port "2379".
In fact, in startEtcd(), we created the etcd server with random port.
So when run test apiserver with "bazel test --config integration //test/integration/apiserver/...", we will see the error below.
```
F0517 09:14:20.409854      14 storage_decorator.go:57] Unable to create storage backend: config (&{ 7e13b177-3dd4-47ef-8c8e-e00670128eb3 {[http://127.0.0.1:2379]   } true 0xc00e568090 <nil> <nil> 5m0s 1m0s}), err (dial tcp 127.0.0.1:2379: connect: connection refused)
```